modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/unified_query_builder/QueryBuilderTab.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayLayout from '@clayui/layout';
+import {PropTypes} from 'prop-types';
+import React, {useEffect} from 'react';
+
+import {
+	SIDEBAR_STATE,
+	getStorageAddSXPElementSidebar,
+} from '../../utils/sessionStorage';
+import {SIDEBAR_TYPES} from '../../utils/types/sidebarTypes';
+import QuerySettings from '../query_builder_tab/QuerySettings';
+import QuerySXPElements from './QuerySXPElements';
+import ScopeSelector from './ScopeSelector';
+import CustomPanel from './shared/CustomPanel';
+
+function QueryBuilderTab({
+	applyIndexerClauses,
+	clauseContributorsList = [],
+	elementInstances,
+	entityJSON,
+	errors = [],
+	frameworkConfig = {},
+	isSubmitting,
+	indexFields,
+	isIndexCompany,
+	onApplyIndexerClausesChange,
+	onBlur,
+	onChange,
+	onDeleteSXPElement,
+	onFetchSearchableTypes,
+	onFrameworkConfigChange,
+	onAssetSubtypesMapChange,
+	scope,
+	searchableTypes = [],
+	setFieldTouched,
+	setFieldValue,
+	openSidebar,
+	setOpenSidebar,
+	setScope,
+	assetSubtypesMap,
+	touched = [],
+}) {
```

Comment:
If this is a new file, it'd be a good idea to convert this to typescript so we can have better defined types when creating the unified query builder.

It'll make the object types more intentional. For example it'll be easier making the `errors` type consistent throughout each component. Also it'll make it quicker to understand what `errors` is since looking at this, I have no idea what data `errors` has.

So by making the first top-level components use typescript, it'll make other components you build stay consistent with their data types.